### PR TITLE
Traktor S3: Disable scratch when switching decks to prevent locked scratch issue

### PR DESCRIPTION
**Pull request by evasadakkanalesso**
Hi there,

I have been using the Traktor S3 controller mappings and have come across this sneaky bug. Steps to reproduce below:

1. Start any song you like and perform a spin in whichever direction you feel like
2. quickly swap decks and you will find the previous deck is kept in scratch mode

This happens due to ```TraktorS3.Deck.prototype.checkJogInertia``` function using ```this.activeChannel``` which is now different than the previous active channel. Traktor DJ Pro works the same way as in if you perform a spin in any direction and swap decks it interrupts the current spinning deck. 

Thanks Mixxx Dev for building such an amazing software!!!!

Thanks,
Eva

### CODE REVIEW

**Comment by Be-ing** [comment]
Thanks. I have retargeted this PR to the 2.3 branch. Please [rebase it](https://github.com/mixxxdj/mixxx/wiki/Using%20Git#targeting-another-base-branch).

**Comment by evasadakkanalesso** [comment]
Hi @Be-ing ,

It's done. 

👍

**Comment by Be-ing** [comment]
Thank you. @ywwg wrote the mapping, so I'll leave it to him to review and test these changes.

File: res/controllers/Traktor-Kontrol-S3-hid-scripts.js
Comment (ywwg) [inline res/controllers/Traktor-Kontrol-S3-hid-scripts.js:1800]: instead of this new "previouschannel" value, why not disable scratching at the top of the deckswitchhandler, before the active channel gets updated?  so, disable scratching and then activateChannel.

**Review by ywwg — CHANGES_REQUESTED** [review]

File: res/controllers/Traktor-Kontrol-S3-hid-scripts.js
Comment (evasadakkanalesso) [reply res/controllers/Traktor-Kontrol-S3-hid-scripts.js:1800]: hi @ywwg - that's doable and also an easier way of doing it. Will make the change 👍 
thank you

**Review by ywwg — APPROVED** [review]

**Review by ywwg — APPROVED** [review]

**Comment by uklotzde** [comment]
@evasadakkanalesso As a *first-time contributor* we have to ask you to sign our contributor agreement: https://docs.google.com/forms/d/e/1FAIpQLScC9QG327sjLL0eWftmfGUasxFFLxg6LCXJ2xHDYRzFIRqyiw/viewform?formkey=dEpYN2NkVEFnWWQzbkFfM0ZYYUZ5X2c6MQ

@ywwg Please check before merging PRs.

**Comment by ywwg** [comment]
ah, yes, sorry.

**Comment by ywwg** [comment]
@evasadakkanalesso Hi, please let us know when you've signed the agreement -- otherwise we'll have to roll back this change because we need every contribution logged.

**Comment by evasadakkanalesso** [comment]
Hi Owen,

I have done it a few days. I thought you'd get a message or something.

Thanks

**Comment by ywwg** [comment]
Thanks!

**Comment by evasadakkanalesso** [comment]
no problem thank you! ♥️

**Pull request merged**
